modules/core/src/mathfuncs_core.simd.hpp

```diff
@@ -257,6 +260,71 @@ void magnitude64f(const double* x, const double* y, double* mag, int len)
     }
 }
 
+void magnitudeSqr32f(const float* x, const float* y, float* mag, int len)
+{
+    CV_INSTRUMENT_REGION();
+
+    int i = 0;
+
+#if CV_SIMD || CV_SIMD_SCALABLE
+    const int VECSZ = VTraits<v_float32>::vlanes();
+    for( ; i < len; i += VECSZ*2 )
+    {
+        if( i + VECSZ*2 > len )
+        {
+            if( i == 0 || mag == x || mag == y )
+                break;
+            i = len - VECSZ*2;
+        }
+        v_float32 x0 = vx_load(x + i), x1 = vx_load(x + i + VECSZ);
+        v_float32 y0 = vx_load(y + i), y1 = vx_load(y + i + VECSZ);
+        x0 = v_muladd(x0, x0, v_mul(y0, y0));
+        x1 = v_muladd(x1, x1, v_mul(y1, y1));
```

Comment:
I tried to use `v_sqr_magnitude` intrinsic and it works well too. Performance on x86_64 is the same at least.